src/realm/array.cpp

```diff
@@ -1717,9 +1717,9 @@ void Array::alloc(size_t init_size, size_t width)
             // field in the header.
             size_t new_capacity_bytes = capacity_bytes * 2;
             if (new_capacity_bytes < capacity_bytes) // overflow detected, clamp to max
```

Comment:
On second thought, why don't we make this change (if any) in a follow up PR as an optimisation, since there could be some discussion about how to do it.